lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchEvaluatorShimConfigurationProvider.java

```diff
@@ -32,10 +38,23 @@
 public class AzureBatchEvaluatorShimConfigurationProvider {
 
   private final RemoteManager remoteManager;
+  private final TcpPortProvider portProvider;
+  private final LocalAddressProvider localAddressProvider;
+  private final Set<String> tcpPortList;
 
   @Inject
-  AzureBatchEvaluatorShimConfigurationProvider(final RemoteManager remoteManager) {
+  AzureBatchEvaluatorShimConfigurationProvider(
+      @Parameter(TcpPortList.class) final Set<Integer> tcpPortList,
+      final RemoteManager remoteManager,
+      final LocalAddressProvider localAddressProvider,
+      final TcpPortProvider portProvider) {
```

Comment:
Why is there a tcpPortList and a TcpPortProvider? It should be one or the other.

Reply:
TcpPortProvider is just the interface. We are not hard-coding the provider here to any implementation. We just read what is provided to the driver and pass it on to the shim.

---
In reply to: [199969149](https://github.com/sharathmalladi/reef/pull/6#discussion_r199969149) [](ancestors = 199969149)